specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,247 @@
+
+The net is that the Reunion PowerManager API will be a combination of WinRT PowerManager and Win32 PowerSettingRegisterNotification.
+
+
+ <br>
+
+## Examples
+### Subscribing and responding to events
+The power notifications are relatively simple. The app provides a callback function for each notification it wants to receive, and when the event is raised the system invokes that callback. The app could start listening for the notifications at any time; it would most often do this early in its life, perhaps in WinMain - but the AppLifecycle mechanism does not impose any constraints on this, and the app is free to subscribe or unsubscribe at any time.
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    // Optionally, register callbacks for power/battery state changes.
+    PowerManager::BatteryStatusChanged([](IInspectable sender, auto args) 
+        { OnBatteryStateChange(sender, args); });
+    PowerManager::PowerSupplyStatusChanged([](IInspectable sender, auto args) 
+        { OnPowerStateChange(sender, args); });
+    PowerManager::EnergySaverStatusChanged([](IInspectable sender, auto args) 
+        { OnEnergySaverStateChange(sender, args); });
+    PowerManager::RemainingChargePercentChanged([](IInspectable sender, auto args) 
+        { OnRemainingChargePercentChange(sender, args); });
+    PowerManager::RemainingDischargeTimeChanged([](IInspectable sender, auto args) 
+        { OnRemainingDischargeTimeChange(sender, args); });
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Standard Win32 window configuration/creation and message pump:
+    // ie, whatever the app would normally do - nothing new here.
+    LoadStringW(hInstance, IDS_APP_TITLE, szTitle, MAX_LOADSTRING);
+    LoadStringW(hInstance, IDC_CLASSIC, szWindowClass, MAX_LOADSTRING);
+    MyRegisterClass(hInstance);
+    if (!InitInstance(hInstance, nCmdShow))
+    {
+        return FALSE;
+    }
+    MSG msg;
+    while (GetMessage(&msg, nullptr, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+    }
+    return (int)msg.wParam;
+}
+```
+
+In the app-defined callback, the app can do whatever it likes. Typically, it would fetch the current status of the particular feature (that is, the battery status, power supply status, energy saver status, and so on), and use that information to form its decision. For example, if battery is at a low power state and discharging, the app might choose to defer any non-critical background work, in the interests of being a good power/battery citizen.
```

Comment:
Somewhere in here, or one of the other docs, we need a note about the threading model for the callbacks.  As this sample explicitly mixes window threads and WinRT callbacks, we should be clear that the `OnRemaining...` will be invoked on the same thread that handles the incoming system thread. It may be up to the app to send itself a window message to hop back onto the window thread, or use a marshalable callback object so COM magically transitions the call back to this thread's [A]STA.

@jeffstall filed https://github.com/microsoft/xlang/issues/673 which goes into some of the details.